magicctapipe/scripts/lst1_magic/lst1_magic_dl2_to_dl3.py

```diff
@@ -331,137 +518,241 @@ def create_pointing_table(event_table):
+                event_table["pointing_alt"][0].to_value(u.deg), ndmin=1
+            ),
+            "AZ_PNT": u.Quantity(
+                event_table["pointing_az"][0].to_value(u.deg), ndmin=1
+            ),
+        }
+    )
 
     pnt_header = fits.Header()
-    pnt_header['CREATED'] = Time.now().utc.iso
-    pnt_header['HDUCLAS1'] = 'POINTING'
-    pnt_header['OBS_ID'] = np.unique(event_table['obs_id'])[0]
-    pnt_header['MJDREFI'] = np.modf(MJDREF)[1]
-    pnt_header['MJDREFF'] = np.modf(MJDREF)[0]
-    pnt_header['TIMEUNIT'] = 's'
-    pnt_header['TIMESYS'] = 'UTC'
-    pnt_header['TIMEREF'] = 'TOPOCENTER'
-    pnt_header['OBSGEO-L'] = (ORM_LON, 'Geographic longitude of the telescopes (deg)')
-    pnt_header['OBSGEO-B'] = (ORM_LAT, 'Geographic latitude of the telescopes (deg)')
-    pnt_header['OBSGEO-H'] = (ORM_HEIGHT, 'Geographic height of the telescopes (m)')
+    pnt_header["CREATED"] = Time.now().utc.iso
+    pnt_header["HDUCLAS1"] = "POINTING"
+    pnt_header["OBS_ID"] = np.unique(event_table["obs_id"])[0]
+    pnt_header["MJDREFI"] = np.modf(MJDREF)[1]
+    pnt_header["MJDREFF"] = np.modf(MJDREF)[0]
+    pnt_header["TIMEUNIT"] = "s"
+    pnt_header["TIMESYS"] = "UTC"
+    pnt_header["TIMEREF"] = "TOPOCENTER"
+    pnt_header["OBSGEO-L"] = (ORM_LON, "Geographic longitude of the telescopes (deg)")
+    pnt_header["OBSGEO-B"] = (ORM_LAT, "Geographic latitude of the telescopes (deg)")
+    pnt_header["OBSGEO-H"] = (ORM_HEIGHT, "Geographic height of the telescopes (m)")
 
     return pnt_table, pnt_header
 
 
-def dl2_to_dl3(input_file_dl2, input_file_irf, output_dir, config):
+def dl2_to_dl3(input_file_dl2, input_dir_irf, output_dir, config):
     """
     Creates a DL3 data file with input DL2 data and IRF files.
 
     Parameters
     ----------
     input_file_dl2: str
         Path to an input DL2 data file
-    input_file_irf: str
-        Path to an input IRF file
+    input_dir_irf: str
+        Path to a directory where input IRF files are stored
     output_dir: str
         Path to a directory where to save an output DL3 data file
     config: dict
         Configuration for the LST-1 + MAGIC analysis
     """
 
-    config_dl3 = config['dl2_to_dl3']
+    # Load the input IRF files:
+    irf_data, extra_header = load_irf_files(input_dir_irf)
 
-    logger.info('\nConfiguration for the DL3 process:')
-    logger.info(config_dl3)
+    quality_cuts = extra_header.get("QUAL_CUT")
+    irf_type = extra_header.get("IRF_TYPE")
+    dl2_weight = extra_header.get("DL2_WEIG")
 
-    hdus = fits.HDUList([fits.PrimaryHDU(), ])
+    # Load the input DL2 data file:
+    event_table, deadc = load_dl2_data_file(
+        input_file_dl2, quality_cuts, irf_type, dl2_weight
+    )
 
-    # Load the input IRF file:
-    logger.info('\nLoading the input IRF file:')
-    logger.info(input_file_irf)
+    pointing_coszd = np.mean(np.sin(event_table["pointing_alt"]))
+    pointing_az = np.mean(event_table["pointing_az"])
```

Comment:
I realized that we might have  a problem here if the azimuth makes a full 2 pi turn. One could check the .std of the azimuth in two definition -180 - 180 and 0-360 and use the smaller one for calculations (then translate it into the default convention of azimuth).  But we can leave it as it is (maybe with a "FIX ME" written in the code), or just an error if .std is greater than  lets say 20 deg